dev_guide/configmaps.adoc

```diff
@@ -0,0 +1,616 @@
+OpenShift.  ConfigMap can be used to store fine-grained information like individual properties or
+coarse-grained information like entire config files or JSON blobs.
+
+The ConfigMap API resource holds key-value pairs of configuration data that can be consumed in pods
+or used to store configuration data for system components such as controllers.  ConfigMap is similar
+to link:Secrets#secrets), but designed to more conveniently support working with strings that do not
+contain sensitive information.
+
+Let's look at a made-up example:
+
+====
+[source,yaml]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  creationTimestamp: 2016-02-18T19:14:38Z
+  name: example-config
+  namespace: default
+data:
+  example.property.1: hello
+  example.property.2: world
+  example.property.file: |-
+    property.1=value-1
+    property.2=value-2
+    property.3=value-3
+----
+====
+
+The `data` field contains the configuration data.  As you can see, ConfigMaps can be used to hold
+fine-grained information like individual properties or coarse-grained information like the contents
+of configuration files.
+
+Configuration data can be consumed in pods in a variety of ways.  ConfigMaps can be used to:
+
+1.  Populate the value of environment variables
+2.  Set command-line arguments in a container
+3.  Populate config files in a volume
+
+Both users and system components may store configuration data in ConfigMap.
+
+[[creating-configmaps]]
+
+## Creating ConfigMaps
+
+You can use the `kubectl create configmap` command to create configmaps easily from literal values,
+files, or directories.
+
+Let's take a look at some different ways to create a ConfigMap:
+
+[[creating-from-directories]]
+
+### Creating from directories
+
+Say that we have a directory with some files that already contain the data we want to populate a ConfigMap with:
+
+====
+----
+$ ls example-files
+game.properties
+ui.properties
+
+$ cat example-files/game.properties
+enemies=aliens
+lives=3
+enemies.cheat=true
+enemies.cheat.level=noGoodRotten
+secret.code.passphrase=UUDDLRLRBABAS
+secret.code.allowed=true
+secret.code.lives=30
+
+$ cat example-files/ui.properties
+color.good=purple
+color.bad=yellow
+allow.textmode=true
+how.nice.to.look=fairlyNice
+----
+====
+
+The `oc create configmap` command can be used to create a ConfigMap holding the content of each
+file in this directory:
+
+====
+----
+$ oc create configmap game-config --from-file=examples/
+----
+====
+
+When `--from-file` points to a directory, each file directly in that directory is used to populate a
+key in the ConfigMap, where the name of the key is the filename, and the value of the key is the
+content of the file.
+
+Let's take a look at the ConfigMap that this command created:
+
+====
+----
+$ oc describe configmaps game-config
+Name:           game-config
+Namespace:      default
+Labels:         <none>
+Annotations:    <none>
+
+Data
+====
```

Comment:
This legitimate `====` in the configmap output is messing up the example/code blocks. Haven't been able to figure out yet a way to escape this. Maybe the easiest solution for now would be to remove the example block (`====`) around it and just let this output only be in a code block (`----`). Seems to render fine that way, at least with asciibinder (and hopefully on the Customer Portal).